atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -963,15 +1001,29 @@ private ListenableFuture<Map<Cell, byte[]>> getInternal(
                                 SafeArg.of("durationMillis", getMillis));
                     }
 
+                    validateFetchedLessOrEqualToExpected(expectedNumberOfPresentCellsToFetch, fromKeyValueService);
```

Comment:
I was thinking about the `-1` case where because of corruption/bad logic, we cache more things than there should be. This handles it fine, in that you are basically still maybe loading things from the KVS and then definitely failing since size can't be less than zero, though do we want to fail earlier in that case?